service/storage_proxy.hh

```diff
@@ -470,7 +470,7 @@ public:
         }
         return next;
     }
-    void init_messaging_service(migration_manager*);
+    void init_messaging_service(netw::messaging_service&, gms::gossiper&, migration_manager&);
```

Comment:
nit: I'd think towards some better name for this method. Maybe as simple as `start_remote`?